components/ee/ws-scheduler/pkg/scheduler/state.go

```diff
@@ -215,26 +205,49 @@ func (s *State) GetAssignedPods(node *Node) []*Pod {
 	return uniquePods(assignedPods)
 }
 
-func uniquePods(pods []*Pod) []*Pod {
+func uniquePods(pods []*corev1.Pod) []*corev1.Pod {
+	if len(pods) == 0 {
+		return pods
+	}
+
 	type value struct {
-		pod   *Pod
+		pod   *corev1.Pod
 		index int
 	}
 	uniqueKeys := make(map[string]value)
-	uniqueList := []*Pod{}
+	var uniqueList []*corev1.Pod
 	for i := 0; i < len(pods); i++ {
 		p := pods[i]
-		existingPod, present := uniqueKeys[p.Pod.Name]
+		existingPod, present := uniqueKeys[p.Name]
+
 		if !present {
-			uniqueKeys[p.Pod.Name] = value{
+			uniqueKeys[p.Name] = value{
 				pod:   p,
 				index: len(uniqueList),
 			}
 			uniqueList = append(uniqueList, p)
-		} else if existingPod.pod.Pod.Generation < p.Pod.Generation {
-			// the other pod is newer than the one already in the uniqueList: take that!
-			existingPod.pod = p
-			uniqueList[existingPod.index] = p
+			continue
+		}
+		if existingPod.pod.Generation != 0 || p.Generation != 0 {
+			// we first had to check if any generation field was non-zero, because the Generation field
```

Comment:
Ouch. Did you see this happen in GKE? :-o

Reply:
Just read:
```
In the official one, at least in the version we're using, that field is not present. 
```
Ouch, this must have changed. What's bad is that the code never checked for the field being present in the first place but I'm 99% it was present in the Kubernetes version I wrote this for originally.